### PR TITLE
Seal internal and private classes in corelib

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to https://github.com/dotnet/runtime/issues/49944

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

File: src/coreclr/System.Private.CoreLib/Tools/GenUnicodeProp/TableLevels.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 namespace GenUnicodeProp
 {
-    internal class TableLevels
+    internal sealed class TableLevels
```
Comment (GrabYourPitchforks) [inline src/coreclr/System.Private.CoreLib/Tools/GenUnicodeProp/TableLevels.cs:8]: Heads up that files like this aren't shipping code. This change is ok, but we'll want to make sure the analyzer rule is scoped only to shipping assemblies.

File: src/coreclr/System.Private.CoreLib/src/System/Diagnostics/StackFrameHelper.cs [hunk 2]
```diff
@@ -152,7 +152,7 @@ internal void InitializeSourceInfo(int iSkip, bool fNeedFileInfo, Exception? exc
             }
         }
 
-        public virtual MethodBase? GetMethodBase(int i)
+        public MethodBase? GetMethodBase(int i)
```
Comment (GrabYourPitchforks) [inline src/coreclr/System.Private.CoreLib/src/System/Diagnostics/StackFrameHelper.cs:155]: I think this class is used by the unmanaged reflection stack. We should double check that changing the type's vtable layout won't break anything.

**Review by GrabYourPitchforks — APPROVED** [review]

**Comment by stephentoub** [comment]
Sealing __Canon broke one ConcurrentBag test that tried to CopyTo an Exception[] into an InvalidOperationException[]... the test expected a cast exception, but in release, it didn't get one.

File: src/coreclr/System.Private.CoreLib/src/System/Diagnostics/StackFrameHelper.cs (hunk 2 above)
Comment (stephentoub) [reply src/coreclr/System.Private.CoreLib/src/System/Diagnostics/StackFrameHelper.cs:155]: Thanks :)

File: src/coreclr/System.Private.CoreLib/Tools/GenUnicodeProp/TableLevels.cs (hunk 1 above)
Comment (stephentoub) [reply src/coreclr/System.Private.CoreLib/Tools/GenUnicodeProp/TableLevels.cs:8]: Yup

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ISymWrapperCore.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ namespace System.Reflection.Emit
     // SymWrapperCore is never instantiated and is used as an encapsulation class.
     // It is our "ISymWrapper.dll" assembly within an assembly.
     //------------------------------------------------------------------------------
-    internal class SymWrapperCore
+    internal sealed class SymWrapperCore
```
Comment (KalleOlaviNiemitalo) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ISymWrapperCore.cs:31]: Should SymWrapperCore be made a static class?

Comment (stephentoub) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ISymWrapperCore.cs:31]: Seems like it probably can be.

Comment (marek-safar) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ISymWrapperCore.cs:31]: I think there is CA1052 rule for that ;-)

Comment (stephentoub) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ISymWrapperCore.cs:31]: I postponed turning that on in the past due to the number of changes required, but I'll take another look at it.

**Pull request merged**